node/src/components/contract_runtime/operations.rs

```diff
@@ -195,21 +195,13 @@ pub fn execute_finalized_block(
                     continue;
                 }
                 TransferResult::Success {
-                    post_state_hash,
-                    transfers,
-                    ..
+                    effects, transfers, ..
                 } => {
-                    // ideally we should be doing something with the transfer records and effects,
-                    // but currently this is auto-commit and if we include the effects in the
-                    // collection they will get double-committed. there's also
-                    // the problem that the execution results type needs to be
-                    // made more expressive / flexible. effects? transfers?
-                    state_root_hash = post_state_hash;
                     let artifact = ExecutionArtifact::new(
                         transaction_hash,
                         transaction.header(),
                         ExecutionResult::V2(ExecutionResultV2::Success {
-                            effects: Effects::new(), // auto commit
+                            effects,
```

Comment:
Shouldn't these effects be committed?

Reply:
indeed...and on that note it seems like tracking artifacts and effects separately could be collapsed...the effects at the end could be extracted from the artifacts.